### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `attr()` of every properties

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8779]
### What minimal example or steps are needed to reproduce the bug?

```css
[data-cursor] {
	cursor: attr(data-cursor type(<custom-ident>), default);
}

[data-anchor] {
	anchor-name: attr(data-anchor type(<custom-ident>));
}

[data-landmark] {
	view-transition-name: attr(data-landmark type(<custom-ident>));
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "declaration-property-value-no-unknown": true
    }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIDaUCGAXVBaMBXATgM4T4C6ABMADoB2V6eRJ85G6+AFGpjgcfuXQBPAA4wOAHjyF0EALZYAlrBroAfAEoANOVgAzVLgA26DQG5aAX1q0UGbKhpgAFiQrU6mJ6-xYaqORgWNk5uB28SQVFxKVwZeSUVdQ1zKxsaOx4jRyg5VHwAa3cbdAA3RRgAdyx2R0JFdEUIGj8AoNZ0di57LGyaXPyCqLFJaVkFZRhVTVSaSxAtED1FIxgAOTaEEBgADwCRVYA6MEJCBfBm5YBzLY9ye-IqEHxjGEInljuH76fYMGz8Bgmi0RPgIGJ8MIsKVUEZcDA-BAsLgaAUaBBKnQEIIXjBaN9rHNzpAaNcAGIkPLoLYAK2INHOsBEZ0QXyeMiEqyMilUH0eIGy6De6CeWnx-I5XJ56Bwl0UVywMhy+SgfKeguFTysIEsQA)

### Which Stylelint-related dependencies are you using?

```json
{
	"stylelint": "16.24.0"
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problems were reported:

```
example.css
2:10-57 error Cannot parse property value "attr(data-cursor type(<custom-ident>), default)" for property "cursor" (declaration-property-value-no-unknown)
6:15-53 error Cannot parse property value "attr(data-anchor type(<custom-ident>))" for property "anchor-name" (declaration-property-value-no-unknown)
10:24-64 error Cannot parse property value "attr(data-landmark type(<custom-ident>))" for property "view-transition-name" (declaration-property-value-no-unknown)
```

### Additional Info

For `content` property, it only supports the fallback value and the unit, but does not support the type. For example:
```css
::after {
	/* The following rules are valid. */
	content: attr(data-content, "fallback");
	content: attr(data-content raw-string);

	/* The following rules will report errors. */
	content: attr(data-content type(<string>));
	content: attr(data-content type(*));
}
```

### Do you have a proposal to fix the bug?

Update syntax in CSSTree.

Sorry, I don't know how to extend the syntax myself using the languageOptions configuration property to support new syntax of `attr()` of every properties. Can you help me?

**Comment by ryo-manba** [comment]
@otomad Thanks for the request and for using the template.

The error message shows that parsing itself is failing:

```shell
Cannot parse property value "attr(data-content type(<string>))" for property "content"
```

Even if you extend the syntax with `languageOptions`, it won't work because the parser fails before syntax validation. 
Currently, `ignoreProperties` is the only workaround.

```json
{
  "rules": {
    "declaration-property-value-no-unknown": [true, {
      "ignoreProperties": {
        "/.+/": ["/^attr\\(/"]
      }
    }]
  }
}
```

See: [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIDaUCGAXVBaMBXATgM4T4C6ABMADoB2595eRJ85G6+AFGpjgcfnLoAngAcYnADx5C6CAFssAS1g10APgCUAGnKwAZqlwAbdJoDctAL61aKDNlQ0wACxIVqdBk9cksNVHkYVnYuHkdnN0ERcSkZOUUVGDUtC2tbGnteYycoeVR8AGsPWgZyADclGAB3LA4nQiV0JQgaf0DgtnQObgcsHJo8gsKhMQlpXFkFZVUNTTSaGxpaeHhUfXQYQU8ygHoAKnIAFRcYcn0IY2MIaqUaAHNyfBMYQjZ8M-LUYxUAOnJ9rtSgxIGpkugQt0wn1QZs1LoqCBDFcAEaoMCFRELMqw8GQnrhHCtOHoJ6oWqyfB3e4LYH0A7HU7nS7XW4PJ4vN63K5PGCiEikrb4EiEf6AumMYl4roEmFStSjWKSSnU1KWLz0XFqfHQ3ha0kxCT7ebqpa0VCUSXXfCsC4QczkKwgbRIpTGGAAOQ6CBAMAAHoFRO7fmBCIRneBWvolPcfTtyIjnu7CIjWPH6IjYGAcvgMC02qJheJ8CIsF9jLgYP4IFhcDRCjQbssEOQkBxK7p0wxETHGx8AApFrbNV6pygSsqI3a-ADUQJbSCnAD1QlQ11ROPPSBPHRKrNuvEsnS7QdH7gAxEj5dA+gBWxBoEdgonDiHjiNkwndPzUY8ROU2WREW0YEPxEb87nQIkaDPLBZFyAooD-EAANedBEWsEArCAA)


This ignores all `attr()` functions since CSSTree's parser cannot handle the new syntax.
**Note**: This also disables validation for all `attr()` usage, including valid cases like `content: attr(data-text)`.

**Comment by ryo-manba** [comment]
I submitted a PR to mdn-data.
https://github.com/mdn/data/pull/1015

**Comment by otomad** [comment]
> I submitted a PR to mdn-data. [mdn/data#1015](https://github.com/mdn/data/pull/1015)

Thanks, and in my testing, `if()` function and custom function `--*()` will also parse error, I will add them to "ignoreProperties".

For someone who want to use it:
```json
{
	"declaration-property-value-no-unknown": [true, {
		"ignoreProperties": {
			"/.+/": ["/(^|[^\\w-])(attr|if|--[\\w-]+)\\(/"]
		}
	}]
}
```
